SendBonusActivationActivity/index.ts

```diff
@@ -38,7 +41,26 @@ const fetchWithTimeout = setFetchTimeout(
 
 const httpFetch = toFetch(fetchWithTimeout);
 
-const adeClient = ADEClient(adeServiceEndpoint, httpFetch);
+const logHttpFetch = (f: typeof fetch): typeof fetch => {
+  return (input, init) => {
+    const res = f(input, init);
+    res
+      .then(_ => _.clone())
+      .then(_ => _.json())
+      .then(_ =>
+        traceAdeRequest({
+          Id: _?.result?.codiceBuono || new Date().getTime().toString(),
+          RequestPayload: init?.body?.toString() || "",
+          ResponsePayload: _,
+          Timestamp: new Date()
+        })
+      )
+      .catch(constVoid);
+    return res;
+  };
+};
+
+const adeClient = ADEClient(adeServiceEndpoint, logHttpFetch(httpFetch));
```

Comment:
Should we put this in `utils/adeClient`?

Reply:
yes it's an early draft just to give the idea. Ideally we should use the same method (intercepting fetch) with the call to INPS as well. I'm not satisfied with the code since now it's all into loggers.ts

can you try do a refactor?

Reply:
since the code to initialize the fetch request is very similar between ade and inps maybe we can put it in a fetch module that returns a configurable  fetch where you can provide a logger impl